prover/crypto/vortex/commitment.go

```diff
@@ -40,15 +39,17 @@ func (p *Params) CommitMerkle(ps []smartvectors.SmartVector) (encodedMatrix Enco
 	timeEncoding := profiling.TimeIt(func() {
 		encodedMatrix = p.encodeRows(ps)
 	})
-
 	timeSisHashing := profiling.TimeIt(func() {
-		colHashes = p.hashColumns(encodedMatrix)
+		// colHashes stores concatenation of SIS+MiMC hashes of the columns
+		// if isSISAppliedForRound is true, otherwise it stores the MiMC hashes
+		// of the columns.
+		colHashes = p.hashColumns(encodedMatrix, isSISAppliedForRound)
 	})
 
 	timeTree := profiling.TimeIt(func() {
 		var leaves []types.Bytes32
 
-		if !p.HasSisReplacement() {
+		if !p.HasSisReplacement() && isSISAppliedForRound {
```

Comment:
When you have a boolean flag that represent an option (which is a priori, not the default behaviour). Then, the practice is it make it false in the default case and true in the optional case.

Since the default behaviour is to use SIS, your flag should be "isSisReplacedByMiMC", so the contrary of that.

That way, all pieces of code that don't want to set this value won't have anything to change